io/net/virt-net/network_virtualization.py

```diff
@@ -528,10 +523,84 @@ def test_vnic_dlpar(self):
+        """
+        for slot_no, device_ip, netmask, mac, peer_ip, sriov_port, adapter_id in zip(self.slot_num,
+                                                                                     self.device_ip,
+                                                                                     self.netmask,
+                                                                                     self.mac_id,
+                                                                                     self.peer_ip,
+                                                                                     self.sriov_port,
+                                                                                     self.backing_adapter_id):
+            self.update_backing_devices(slot_no)
+            device_name = self.find_device(mac)
+            networkinterface = NetworkInterface(device_name, self.local)
+            count = 0
+            self.log.info("Preforming DLPAR on %s" % device_name)
+            for _ in range(self.num_of_dlpar):
+                self.log.info("DLPAR iteration #%d" % count)
+                num_backingdevs = self.backing_dev_count_w_slot_num(slot_no)
+
+                self.device_add_remove(slot_no, '', '', '', 'remove')
+                if networkinterface.is_available():
+                    self.fail("DLPAR remove did not remove interface")
+
+                self.device_add_remove(slot_no, mac, sriov_port, adapter_id, 'add')
+                for c in range(1, num_backingdevs):
+                    self.backing_dev_add_remove('add', c)
+                    self.wait_interface(device_name)
+
+                try:
+                    networkinterface.add_ipaddr(device_ip, netmask)
+                except Exception:
+                    networkinterface.save(device_ip, netmask)
+                    networkinterface.add_ipaddr(device_ip, netmask)
+                networkinterface.bring_up()
+
+                if not wait.wait_for(networkinterface.is_link_up, timeout=120):
+                    self.fail("Unable to bring up the link on the Network virtualized device")
+
+                time.sleep(5)
+
+                if networkinterface.ping_check(peer_ip, count=5) is not None:
+                    self.fail("dlpar has affected Network connectivity")
+                count += 1
+        self.check_dmesg_error()
+
+    def backing_dev_count_w_slot_num(self, slot):
+        """
+        Lists the count of backing devices
+        param slot: vnic slot
+        returns: backing device count
+        """
+        count = 0
+        output = self.backing_dev_list()
+        for i in output.splitlines():
+            if i.startswith('%s,' % slot):
+                count = len(i.split(',')[1:])
+        return count
+
+    def wait_interface(self, device_name):
+        """
+        Wait till interface comes up
+        param device_name: vnic device that is tested
+
+        returns: if the device is up or down
+        """
```

Comment:
@vaishnavibhat Please use the following reference for proper pep standards,

```
def get_dir_mountpoint(dir_path):
    """
    get mounted disk name that is mounted on given dir_path

    :param dir_path: absolute directory path
    :type dir_path: str

    :returns: returns disk name which mounted on given dir_path
    :rtype: str
    """

```

Reply:
Resent patch with the above format.